src/GafferAppleseed/IECoreAppleseedPreview/Renderer.cpp

```diff
@@ -1693,6 +1713,201 @@ class AppleseedDeltaLight : public AppleseedLight
 		asf::Transformd m_transform;
 };
 
+/// Appleseed area light handle.
+class AppleseedAreaLight : public AppleseedLight
+{
+	public :
+
+		AppleseedAreaLight( asr::Project &project, const string &name, const IECoreScenePreview::Renderer::AttributesInterface *attributes, IECoreScenePreview::Renderer::RenderType renderType )
+			:	AppleseedLight( project, name, attributes, renderType == IECoreScenePreview::Renderer::Interactive ) , m_transform( asf::Transformd::identity() ), m_renderType( renderType )
+		{
+			init();
+			AppleseedAreaLight::attributes( attributes );
+		}
+
+		virtual ~AppleseedAreaLight()
+		{
+			if( isInteractiveRender() )
+			{
+				removeAreaLightEntities();
+			}
+			else
+			{
+				// Create the material assignments.
```

Comment:
The rect area light is a primitive grid with a diffuse_edf assigned. 
I need to create the object instance in the destructor for non-interactive renderers because the 
ObjectInstance class is not mutable. If I create it in attributes() I cannot update the transformation matrix when transform() is called.